### PR TITLE
Warn for more bad practice

### PR DESCRIPTION
**Pull request by mzuenni**
We could warn for even more stuff defined in `random` but let's at least warn for those methods implemented in `validation.h`.

### CODE REVIEW

**Comment by RagnarGrootKoerkamp** [comment]
Nice!

Do you think we really need the `typeid()`? I don't think that's likely to be used for generators right?

Should we check this for all programs though? I think we only want to forbid these things for generators and (interactive) validators. Team submissions are fine

**Comment by mzuenni** [comment]
well it already warned for `rand()` I just added a few more... I am not sure for which programs this is actually checked...

Probably no one uses typeid but it's compiler dependent and it won't harm I guess?

File: bin/program.py [hunk 1]
```diff
@@ -287,19 +287,35 @@ def _checks(self):
             if self.language == 'cpp':
```
Comment (RagnarGrootKoerkamp) [inline bin/program.py:287]: can you add `or (self.problem.interactive and isinstance(self, Validator)` and test that?  (I'm not exactly sure whether that `self.problem.interactive` is exactly right.)

Comment (mzuenni) [reply bin/program.py:287]: why not just warn for all validators? I don't see why a normal output validator should be allowed to be non-deterministic but in an interactive problem, we don't want to allow it?

Comment (RagnarGrootKoerkamp) [reply bin/program.py:287]: ok let's do all and revisit when people complain ;)

**Pull request merged**